### PR TITLE
fix(server):set convict to strict validation

### LINKED ISSUE #2507: set convict to strict validation
Stealing @seanmonstar's excellent https://github.com/mozilla/fxa-oauth-server/issues/263 issue

---

> _"This will catch when any config value is added that shouldn't have been, or typos. Requires a bump to convict@0.8"_

### PR DESCRIPTION
**Pull request by TDA**
Fixes #2507 
@zaach

### CODE REVIEW

File: pm2 [hunk 1]
```diff
@@ -0,0 +1 @@
+node_modules/.bin/pm2
```
Comment (zaach) [inline pm2]: Remove this.

File: package.json [hunk 2]
```diff
@@ -31,7 +31,7 @@
     "connect-fonts-firasans": "0.0.4",
     "connect-restreamer": "1.0.0",
     "consolidate": "0.10.0",
-    "convict": "0.6.0",
+    "convict": "0.8.0",
```
Comment (zaach) [inline package.json]: We'll need to build a new shrinkwrap file.

Comment (vladikoff) [inline package.json]: @TDA make sure you are using npm 2.4+ , I suggest npm 2.10.

remove the `npm-shinkwrap.json`
and run `npm shrinkwrap` to create the new file

**Comment by zaach** [comment]
Aha! It's already caught an error:

```
Loading "Gruntfile.js" tasks...ERROR
>> Error: configuration param 'are_dist_resources' not declared in the schema
```

We'll need to add that property to the schema, with a default value, doc, etc.

**Comment by zaach** [comment]
It looks like you've stacked the commits on top of each other. Can you squash these commits into one commit and force push? How to squash: http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

File: pm2 (hunk 1 above)
Comment (zaach) [inline pm2]: This extra file is still here, maybe because of the multiple commits.

**Comment by vladikoff** [comment]
from meeting:

change `chore` to `fix` in the commit message

**Comment by TDA** [comment]
Done :)

**Comment by vladikoff** [comment]
@TDA nice! still need to squash the commits though, see https://github.com/mozilla/fxa-content-server/pull/2510#issuecomment-108729846

image

**Comment by TDA** [comment]
@vladikoff : done :)

**Comment by vladikoff** [comment]
@TDA also need to update npm shrinkwrap, see comment https://github.com/mozilla/fxa-content-server/pull/2510/files#r31582960

**Comment by TDA** [comment]
@vladikoff okie, done.

**Comment by vladikoff** [comment]
```
The command "grunt validate-shrinkwrap --force" exited with 0.
3.41s$ grunt lint
Loading "Gruntfile.js" tasks...ERROR
>> Error: configuration param 'are_dist_resources' not declared in the schema
Warning: Task "lint" not found. Use --force to continue.
Aborted due to warnings.
```

**Comment by zaach** [comment]
@TDA It looks like during the rebase you lost the modification to the schema that added the `are_dist_resources` property.

**Comment by TDA** [comment]
ok, added that back in.

File: server/lib/configuration.js [hunk 3]
```diff
@@ -147,6 +147,11 @@ var conf = module.exports = convict({
     format: Array,
     default: []
   },
+  are_dist_resources: {
+    doc: 'Check if the resources are under the /dist directory',
+    format: Boolean,
+    default: false
```
Comment (vladikoff) [inline server/lib/configuration.js]: @zaach should `false` be default here?

**Comment by vladikoff** [comment]
review+ with comment: https://github.com/mozilla/fxa-content-server/pull/2510#discussion_r31780051

File: server/lib/configuration.js (hunk 3 above)
Comment (zaach) [inline server/lib/configuration.js]: This should be okay since we set the value below based on `static_directory`.

**Comment by zaach** [comment]
Thanks @TDA! :+1:

**Comment by TDA** [comment]
:D

**Pull request merged**